server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -4371,11 +4371,22 @@ public void close() throws IOException {
     }
 
     /**
-     * Downloads segments from remote segment store.
+     * Downloads segments from remote segment store. This method will download segments till
+     * last refresh checkpoint.
      * @param overrideLocal flag to override local segment files with those in remote store
      * @throws IOException if exception occurs while reading segments from remote store
      */
     public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal) throws IOException {
+        syncSegmentsFromRemoteSegmentStore(overrideLocal, true);
+    }
+
+    /**
+     * Downloads segments from remote segment store.
+     * @param overrideLocal flag to override local segment files with those in remote store
+     * @param refreshLevelSegmentSync last refresh checkpoint is used if true, commit checkpoint otherwise
+     * @throws IOException if exception occurs while reading segments from remote store
+     */
+    public void syncSegmentsFromRemoteSegmentStore(boolean overrideLocal, boolean refreshLevelSegmentSync) throws IOException {
```

Comment:
Can sync segments be agnostic of the source like peer node or remote store?

Reply:
Looks quite challenging to me as the code paths are completely different . Would like to know @sachinpkale's view as well . 

Also the paths are different in other parts like restore as well . So there is consistency across all parts .

Reply:
As @gbbafna mentioned, this method is not specific to peer recovery. We are using the same method in restore as well as failover flow.

Reply:
In this change, we make sure that segments are downloaded from the remote store instead of the source node. Remote store is not replacing the source node. I have not given much thought to this yet but given the peer recovery flow, it would be much bigger change.

Reply:
The method name itself doesn't provide an abstraction. Lets open a follow up issue to clean this up ?

Reply:
> The method name itself doesn't provide an abstraction. Lets open a follow up issue to clean this up ?

Okay, will create a tracking issue.

Reply:
Created issue: https://github.com/opensearch-project/OpenSearch/issues/6831